airflow/providers/amazon/aws/hooks/s3.py

```diff
@@ -909,7 +919,17 @@ def download_file(self, key: str, bucket_name: str | None = None, local_path: st
                 Config=self.transfer_config,
             )
 
-        return local_tmp_file.name
+        if preserve_file_name:
+            filename_in_s3 = s3_obj.key.split('/')[-1]
+            local_folder_name = local_tmp_file.name.rsplit('/', 1)[0]
+            local_file_name = f"{local_folder_name}/{filename_in_s3}"
+
+            self.log.info("Renaming file from %s to %s", local_tmp_file.name, local_file_name)
+            rename(local_tmp_file.name, local_file_name)
+
+            return local_file_name
```

Comment:
```suggestion
            filename_in_s3 = s3_obj.key.rsplit('/', 1)[-1]
            local_tmp_file_name = Path(local_tmp_file.name)
            local_file_name = local_tmp_file_name.with_name(filename_in_s3)

            self.log.info("Renaming file from %s to %s", local_tmp_file_name, local_file_name)
            local_tmp_file_name.rename(local_file_name)

            return str(local_file_name)
```

Don’t do string manipulation on the path, use structured calls instead.

Reply:
Thank you for your comment, but I've refactored the code according to @o-nikolas's suggestion, so now this is not relevant :( 

@uranusjr  - Can you please review the new implementation? 🙏 It's a bit simpler, and I made sure that I'm using structured calls for paths.